### PR TITLE
update(rtlcss): v3.1 (postcss 8)

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- version bump
- postcss v8 support
- configuration for newer types in postcss dependnecy
- tests amended

https://github.com/MohammadYounes/rtlcss/compare/2.6.2...3.1.2

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/rtlcss/rtlcss-tests.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
-import * as rtlcss from "rtlcss";
-import { Transformer } from 'postcss';
+import rtlcss = require('rtlcss');
```
Comment (adamzerella) [inline types/rtlcss/rtlcss-tests.ts:1]: Any reason why we're changing the import syntax here?

File: types/rtlcss/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
     "compilerOptions": {
       "module": "commonjs",
       "lib": [
-        "es6"
+        "es6",
+        "ES2018.Promise"
```
Comment (adamzerella) [inline types/rtlcss/tsconfig.json:6]: Are we using this lib anywhere?

**Review by azerella — CHANGES_REQUESTED** [review]
Couple minor things, everything else LGTM 👍🏻

File: types/rtlcss/rtlcss-tests.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/rtlcss/rtlcss-tests.ts:1]: Oh, no particular one, just intent to align with now recommended syntax:
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
it's for DT, not for consumers of the package,

File: types/rtlcss/tsconfig.json (hunk 2 above)
Comment (peterblazejewicz) [reply types/rtlcss/tsconfig.json:6]: Nope, it's required for DT linter only, and the 'tsconfig' is for DT consumption as well.
The problem is postcss 8 comes with modern language features, including 'finally' support for Promise, 
so as external lib, that is a way to suppress the TS error for extenral dependency.
```bash
Error: Errors in typescript@4.2 for external dependencies:
node_modules/postcss/lib/lazy-result.d.ts(60,28): error TS2339: Property 'finally' does not exist on type 'Promise<Result>'.
```

File: types/rtlcss/rtlcss-tests.ts (hunk 1 above)
Comment (adamzerella) [reply types/rtlcss/rtlcss-tests.ts:1]: Oh no worries, good pickup :)

**Review by azerella — APPROVED** [review]
LGTM!

**Review by rbuckton — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💋

**Pull request merged**